### ISSUE TITLE
Build issue

### ISSUE THREAD

**Issue by cnpetra** [LLNL/hiop#46]
Two reports of build failure came in the last week via email. Coincidentally, I've just encountered the same exact problem on summit (using cmake 3.9.2)

```cmake ..
CMake Error at CMakeLists.txt:7 (cmake_policy):
  Policy "CMP0074" is not known to this version of CMake.


-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
(...)
-- Found LAPACK libraries: /autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-9.1.0/openblas-0.3.9-aymovpat33osbzgh5gsmhyvstsol4sfp/lib/libopenblas.so;/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-9.1.0/openblas-0.3.9-aymovpat33osbzgh5gsmhyvstsol4sfp/lib/libopenblas.so
CMake Error at src/Optimization/CMakeLists.txt:2 (target_link_libraries):
  Object library target "hiopOptimization" may not link to anything.


-- Configuring incomplete, errors occurred!
See also "/ccs/home/cpetra/work/projects/hiop/build/CMakeFiles/CMakeOutput.log".
See also "/ccs/home/cpetra/work/projects/hiop/build/CMakeFiles/CMakeError.log".
```

@ashermancinelli

**Comment by cnpetra** [comment]
but 

```
module load cmake/3.15.2
rm -f *; cmake ..
```

works

**Comment by ashermancinelli** [comment]
I see. We manually enable a policy to suppress warnings, but we should have checked that the minimum required cmake supported that policy. I'll have a PR out soon.

**Comment by pelesh** [comment]
I think this will do it:
```cmake
if (POLICY CMP0074)
  cmake_policy(SET CMP0074 NEW) # CMake 3.12
endif ()
```

**Comment by ashermancinelli** [comment]
It seems we have other issues building with older cmake versions as well. Is it too much to ask to make cmake/12 our lowest supported version?

**Comment by cnpetra** [comment]
cmake min version 3.12 is a bit too constraining at this time...

**Comment by cnpetra** [comment]
after weighing the pros and cons with @pelesh let go with min ver 3.12

**Comment by ashermancinelli** [comment]
I believe this is resolved with #57

**Comment by cnpetra** [comment]
correct
